hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/InconsistentAmazonS3Client.java

```diff
@@ -426,8 +490,10 @@ private void registerDeleteObject(String key, String bucket) {
     if (shouldDelay(key)) {
       // Record summary so we can add it back for some time post-deletion
       S3ObjectSummary summary = null;
-      ObjectListing list = listObjects(bucket, key);
-      for (S3ObjectSummary result : list.getObjectSummaries()) {
+      ListObjectsV2Request request =
+          new ListObjectsV2Request().withBucketName(bucket)
+              .withPrefix(key);
```

Comment:
Any reason for the change here? Do we need to skip all the tests that depend on this class when v1 API is configured?

Reply:
No real reason. I do recall how an aggressive throttle rate had real problems with listing because a single op could fail too many times for the retry handlers to cope. But I don't see that this has fixed it because its not calling the innerList() op which is needed to bypass that inconsistency. 
I'll revert to v2 and innerListV1, so that there's no failures in this loop, and, assuming all stores support v1 alongside v2, work everywher

Reply:
moved back to v1. IDE also suggested fancy java 8 findfirst() streaming op, which I also switched to. Once you can read it, it's better